packages/bonito-core/src/form/validation-status.ts

```diff
@@ -1,13 +1,15 @@
 /**
  * Represents the result of a given validation
+ * VD default to any, to have better type compatibility as it
+ * can be initialized with without a data object and assigned
+ * to a ValidationSnapshot with a data type.
  */
-export class ValidationStatus {
-    level: "ok" | "warn" | "error" | "canceled";
-    message?: string;
+export class ValidationStatus<VD extends undefined | any = any> {
     forced?: boolean = false;
 
-    constructor(level: "ok" | "warn" | "error" | "canceled", message?: string) {
-        this.level = level;
-        this.message = message;
-    }
+    constructor(
+        public level: "ok" | "warn" | "error" | "canceled",
+        public message?: string,
+        public data?: VD
```

Comment:
What do you think about VD meaning ValidationDetails instead of ValidationData? It's not much more specific, but somewhat matches Batch's error pattern of message + details.

Reply:
Like it, change to ValidationDetials